co-simulation/fed/mosaic-carla/src/main/java/org/eclipse/mosaic/fed/carla/carlaconnect/CarlaXmlRpcClient.java

```diff
@@ -63,33 +71,35 @@ public void initialize(URL xmlRpcServerUrl)
     }
 
 
-    /**
-    * This method uses xmlrpc to connect with CARLA CDASim adapter service
-     * @throws XmlRpcException
-    */
-    public void requestCarlaList()
-    {
-        if(!isConnected)
-        {
-            
-            try {
-                throw new XmlRpcException("Server is not connected!");
-            } 
-            catch (XmlRpcException e) 
-            {
-                log.error("Server is not connected! {}", e.getMessage());
-            }
-        }
-        try{          
-            Object[] params = new Object[]{"Test " + java.time.LocalDateTime.now()};
-            Object result = client.execute(registeredFunction, params);
+    public void createSensor(DetectorRegistration registration) throws XmlRpcException{
+        List<Double> location = Arrays.asList(registration.getDetector().getLocation().getX(), registration.getDetector().getLocation().getY(), registration.getDetector().getLocation().getZ());
+        List<Double> orientation = Arrays.asList(registration.getDetector().getOrientation().getPitch(), registration.getDetector().getOrientation().getRoll(), registration.getDetector().getOrientation().getYaw());
+
+        if (isConnected) {
+            Object[] params = new Object[]{registration.getInfrastructureId(), registration.getDetector().getSensorId(), location, orientation};
+            Object result = client.execute(CREATE_SENSOR, params);
             log.info((String)result);
         }
-        catch (XmlRpcException x) 
-        {
-            log.error("Errors occurred with xmlrpc connection {}", x.getMessage());
-            closeConnection();
-        } 
-        
+        else {
+            log.warn("XMLRpcClient is not connected to CARLA Adapter!");
+        }
     }
+
+    public DetectedObject[] getDetectedObjects(String infrastructureId ,String sensorId) throws XmlRpcException{
+        if (isConnected) {
+            Object[] params = new Object[]{infrastructureId, sensorId};
+            Object result = client.execute(GET_DETECTED_OBJECTS, params);
+            log.info("Detections from infrastructure {} sensor {} : {}", infrastructureId, sensorId, result);
```

Comment:
I am thinking this may get really big log if we detect the log the result continuously? Maybe we should move it into debug?

Reply:
I can move this to debug.